### PR TITLE
add fancy-collections

### PR DESCRIPTION
**Pull request by bernt-matthias**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bernt-matthias** [comment]
requires https://github.com/conda-forge/staged-recipes/pull/23087

File: recipes/fancy-collections/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - fancy_collections
+
+about:
+  home: https://github.com/palmb/fancy-collections/
+  summary: Useful classes to store data in a structured way
+  description: |
+    This package provide some useful classes to store data in a structured way.
+    Up to now mostly advanced dictionaries are provided.
+  license: BSD-3-Clause
+  license_family: GPL
```
Comment (bgruening) [inline recipes/fancy-collections/meta.yaml:38]: Here is again some inconsistency ...

File: recipes/fancy-collections/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  imports:
+    - fancy_collections
+
+about:
+  home: https://github.com/palmb/fancy-collections/
+  summary: Useful classes to store data in a structured way
+  description: |
+    This package provide some useful classes to store data in a structured way.
+    Up to now mostly advanced dictionaries are provided.
+  license: BSD-3-Clause
```
Comment (bernt-matthias) [inline recipes/fancy-collections/meta.yaml:37]:
```suggestion
  license: GPL-2.0-or-later
```

File: recipes/fancy-collections/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  run:
+    - python >=3.7
+    - numpy
+    - scipy
+    - pandas
+    - sliceable-dict
+
+test:
+  imports:
+    - fancy_collections
```
Comment (bernt-matthias) [inline recipes/fancy-collections/meta.yaml:30]:
```suggestion
    - fancy_collections
    - fancy_collections.core
    - fancy_collections.formatting
    - fancy-collections. lib
```

File: recipes/fancy-collections/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - fancy_collections.core
+    - fancy_collections.formatting
+    - fancy-collections. lib
```
Comment (bernt-matthias) [inline recipes/fancy-collections/meta.yaml:33]:
```suggestion
    - fancy-collections.lib
```

File: recipes/fancy-collections/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - scipy
+    - pandas
+    - sliceable-dict
+
+test:
+  imports:
+    - fancy_collections
+    - fancy_collections.core
+    - fancy_collections.formatting
+    - fancy-collections.lib
```
Comment (bernt-matthias) [inline recipes/fancy-collections/meta.yaml:33]:
```suggestion
    - fancy_collections.lib
```

**Pull request merged**